src/Diagrams/Trail.hs

```diff
@@ -1326,3 +1331,84 @@ instance (Metric v, OrderedField n) => Reversing (Located (Trail' l v n)) where
 instance (Metric v, OrderedField n) => Reversing (Located (Trail v n)) where
   reversing = reverseLocTrail
 
+------------------------------------------------------------
+--  Serialize instances
+------------------------------------------------------------
+
+instance (Serialize (v n), Serialize (V (v n) (N (v n))), OrderedField n, Metric v) => Serialize (Trail v n) where
+  {-# INLINE get #-}
+  get = do
+    isLine <- Serialize.get
+    case isLine of
+      True  -> do
+        segTree <- getSegTree
+        return (Trail (Line segTree))
+      False -> do
+        segTree <- getSegTree
+        segment <- getOpenSegment
+        return (Trail (Loop segTree segment))
+
+  {-# INLINE put #-}
+  put (Trail (Line segTree)) = do
+    Serialize.put True
+    putSegTree segTree
+
+  put (Trail (Loop segTree segment)) = do
+    Serialize.put False
+    putSegTree segTree
+    putOpenSegment segment
+
+{-# INLINE putSegTree #-}
+putSegTree :: (Serialize (v n)) => SegTree v n -> Put
+putSegTree (SegTree xs) = Serialize.Put.putListOf putClosedSegment (F.toList xs)
+
+{-# INLINE getSegTree #-}
+getSegTree :: (Serialize (v n), OrderedField n, Metric v) => Get (SegTree v n)
+getSegTree = (SegTree . FT.fromList) <$> Serialize.Get.getListOf getClosedSegment
```

Comment:
Not really. Earlier it was to avoid Orphan instances, but now since we're in `diagrams-lib` that's no longer a concern. 

I'll go ahead and make the instances.